addons/lunch/models/lunch_order.py

```diff
@@ -41,10 +41,128 @@ class LunchOrder(models.Model):
 
     display_toppings = fields.Text('Extras', compute='_compute_display_toppings', store=True)
 
+    product_description = fields.Text('Description', related='product_id.description')
+    topping_label_1 = fields.Char(related='product_id.category_id.topping_label_1')
+    topping_label_2 = fields.Char(related='product_id.category_id.topping_label_2')
+    topping_label_3 = fields.Char(related='product_id.category_id.topping_label_3')
+    topping_quantity_1 = fields.Selection(related='product_id.category_id.topping_quantity_1')
+    topping_quantity_2 = fields.Selection(related='product_id.category_id.topping_quantity_2')
+    topping_quantity_3 = fields.Selection(related='product_id.category_id.topping_quantity_3')
+    image_1920 = fields.Image(related='product_id.image_1920')
+    image_128 = fields.Image(related='product_id.image_128')
+
+    available_toppings_1 = fields.Boolean(help='Are extras available for this product', compute='_compute_available_toppings')
+    available_toppings_2 = fields.Boolean(help='Are extras available for this product', compute='_compute_available_toppings')
+    available_toppings_3 = fields.Boolean(help='Are extras available for this product', compute='_compute_available_toppings')
+
+    # TODO Try to remove this, at least make it a compute dependant on the context
+    edit = fields.Boolean('Currently in edit mode', compute='_compute_edit')
+
+    def _compute_edit(self):
+        value = bool(self._context.get('active_id', False))
+        for line in self:
+            line.edit = value
+
+    @api.depends('product_id')
+    def _compute_available_toppings(self):
+        for line in self:
+            line.available_toppings_1 = bool(line.env['lunch.topping'].search_count([('category_id', '=', line.category_id.id), ('topping_category', '=', 1)]))
+            line.available_toppings_2 = bool(line.env['lunch.topping'].search_count([('category_id', '=', line.category_id.id), ('topping_category', '=', 2)]))
+            line.available_toppings_3 = bool(line.env['lunch.topping'].search_count([('category_id', '=', line.category_id.id), ('topping_category', '=', 3)]))
+
     def init(self):
         self._cr.execute("""CREATE INDEX IF NOT EXISTS lunch_order_user_product_date ON %s (user_id, product_id, date)"""
             % self._table)
 
+
+    def _extract_toppings(self, values):
+        """
+            If called in api.multi then it will pop topping_ids_1,2,3 from values
```

Comment:
This is a strange method that will most probably break in the future ...
Parsing the orm commands is always a bad idea, but it is necessary here
The method should probably be inlined in the code as it is simple and mutate the `values` arguments in some cases